modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java

```diff
@@ -2145,18 +1997,22 @@ private void _invokeClientJSGenerator(String openAPIYAMLString)
 		FileUtil.write(
 			new File(baseClientJSDir, "node-scripts.config.js"),
 			FreeMarkerUtil.processTemplate(
-				null, null, "node_scripts_config_js", null));
+				null, null, "ts/node_scripts_config_js", null));
 		FileUtil.write(
 			new File(baseClientJSDir, "package.json"),
 			FreeMarkerUtil.processTemplate(
-				null, null, "package_json",
+				null, null, "ts/package_json",
 				HashMapBuilder.<String, Object>put(
 					"clientName", baseClientJSDir.getName()
 				).build()));
 
 		File openAPIYAMLFile = _prepareForClientJSGenerator(openAPIYAMLString);
```

Comment:
Instead of returning a File we can just return the OpenAPI object without writing anything to the file system. Then you can pass it to the generator and remove the delete after it.

Reply:
Wrong line... I meant https://github.com/liferay/liferay-portal/blob/c18c8c91f1b7f35f90fad56ed798cb9614c97fef/modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/RESTBuilder.java#L2009.